weldx/util/util.py

```diff
@@ -247,7 +247,15 @@ def _visit(path, key, value, a, b) -> bool:
            When the elements are not equal traversing `a` will be stopped
            by raising a RuntimeError.
         """
+        data_structure = iterutils.get_path(a, path)
         other_data_structure = iterutils.get_path(b, path)
+
+        # directly test for equality with sets
+        if isinstance(data_structure, set) or isinstance(other_data_structure, set):
+            if data_structure == other_data_structure:
+                return True
```

Comment:
The problem with this is, that it does not perform further nested comparisons. If I have a set of nested structures, this won't work. I am not sure if we would ever have such a data structure, but the function should still perform its task correctly. So we would need to call `compare_nested` on all correct member pairs and also check the container sizes.

Reply:
in that case it's hard to not generate a list (to index the `other_data_structure`). I have updated to code to work on the individual set items instead of `==`
since sets cannot contain mutable objects I'm having a hard time even creating a test case for nested sets here